src/components/admin/UserConfig.js

```diff
@@ -0,0 +1,133 @@
+import React, {Component} from 'react';
+import Select from "react-dropdown-select"
+import {
+    Button,
+    Input,
+    InputGroup,
+    InputGroupAddon,
+    InputGroupText
+} from "reactstrap";
+
+import DB from "../../state/DB";
+
+export default class UserConfig extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = ({
+            users: DB.users,
+            search_value: '',
+            selected_user: null,
+            selected_user_name: null,
+            selected_user_id: null,
+        });
+
+        this.handleSelectUser = this.handleSelectUser.bind(this);
+        this.handleNameChange = this.handleNameChange.bind(this);
+        this.handleIDChange = this.handleIDChange.bind(this);
+        this.handleDropUser = this.handleDropUser.bind(this);
+        this.handleSaveChangesToUser = this.handleSaveChangesToUser.bind(this);
+        this.handleSaveAllChanges = this.handleSaveAllChanges.bind(this);
+    }
+
+    componentWillMount() {
+        this.setState({
+            users: DB.users
+        });
+    }
+
+    handleSaveAllChanges() {
+        DB.setUsers(this.state.users);
+        DB.updateUsersFile();
+    }
+
+    handleSaveChangesToUser() {
+        let user = {
+            ...this.state.selected_user,
+            ...{'name': this.state.selected_user_name},
+            ...{'id': this.state.selected_user_id}
+        };
+
+        let users = this.state.users;
+        users[this.state.users.indexOf(this.state.selected_user)] = user;
+
+        this.setState({
+            users: users,
+        })
+    }
+
+    handleDropUser() {
+        let users = this.state.users.filter(user => user !== this.state.selected_user);
+
+        this.setState({
+            users: users,
+            selected_user: null,
+            selected_user_name: null,
+            selected_user_id: null,
+        })
+    }
+
+    handleNameChange(event) {
```

Comment:
this is kind of weird because you're doing the same input handling different ways in different components. I suggest just

> why don't you just make the name in state the actual name attribute of the element so you can use event.target.name and not complicate this function